solver/src/liquidity_collector.rs

```diff
@@ -4,17 +4,23 @@ use ethcontract::BlockNumber;
 use crate::{
     liquidity::uniswap::UniswapLikeLiquidity, liquidity::Liquidity, orderbook::OrderBookApi,
 };
+use model::order::OrderUid;
+use std::collections::HashSet;
 
 pub struct LiquidityCollector {
     pub uniswap_like_liquidity: Vec<UniswapLikeLiquidity>,
     pub orderbook_api: OrderBookApi,
 }
 
 impl LiquidityCollector {
-    pub async fn get_liquidity(&self, at_block: BlockNumber) -> Result<Vec<Liquidity>> {
+    pub async fn get_liquidity(
+        &self,
+        at_block: BlockNumber,
+        excluded_orders: &HashSet<OrderUid>,
+    ) -> Result<Vec<Liquidity>> {
         let limit_orders = self
             .orderbook_api
-            .get_liquidity()
+            .get_liquidity(excluded_orders)
```

Comment:
nit: I would do the filtering here instead of leaking this logic further down the stack.

Reply:
I will have to check on this again, but noticed this method converts Orders to LimitOrder which no longer seem to have a UId. Will check back on this.